pkg/ktime/ktime_windows.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package ktime
+
+import (
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var (
+	dll            = syscall.MustLoadDLL("kernel32.dll")
+	queryCounter   = dll.MustFindProc("QueryPerformanceCounter")
+	queryFrequency = dll.MustFindProc("QueryPerformanceFrequency")
+)
+
+func ToProto(ktime uint64) *timestamppb.Timestamp {
+	return ToProtoOpt(ktime, true)
+}
+
+func ToProtoOpt(ktime uint64, monotonic bool) *timestamppb.Timestamp {
+	decodedTime, err := DecodeKtime(int64(ktime), monotonic)
+	if err != nil {
+		logrus.WithError(err).WithField("ktime", ktime).Warn("Failed to decode ktime")
```

Comment:
Seems better to return an error here?


As an aside, please  use logger.GetLogger(): https://github.com/cilium/tetragon/blob/84212ddf3a8cb0fe5df2e77fe497f11e859996a5/pkg/logger/log.go#L151
for logging.

Reply:
This function is common between Linux and Windows, only implementation differs. Changing prototype here without changing calling code.

Reply:
I think a follow on PR to change the common code and each implementation to return the error would be fine.